mne/source_estimate.py

```diff
@@ -3101,12 +3162,21 @@ def save_stc_as_volume(fname, stc, src, dest='mri', mri_resolution=False):
         mri_vol = np.zeros(mri_shape)
         interpolator = src[0]['interpolator']
 
-    for k, v in enumerate(vol):
-        v[mask3d] = stc.data[:, k]
-        if mri_resolution:
-            mri_vol[k] = (interpolator * v.ravel()).reshape(mri_shape3d)
+    n_vertices_seen = 0
+    for src_idx in range(len(src)):
```

Comment:
`src_idx` is only used to iterate over `src` so you should just iterate directly like:
```
for src_part in src:
    assert tuple(src_part['shape'] ...)
```